src/expr/src/relation/func.rs

```diff
@@ -787,12 +787,22 @@ impl AggregateFunc {
             // Note AggregateFunc::MaxString, MinString rely on returning input
             // type as output type to support the proper return type for
             // character input.
-            _ => input_type.scalar_type,
+            _ => input_type.scalar_type.clone(),
         };
         // Count never produces null, and other aggregations only produce
         // null in the presence of null inputs.
         let nullable = match self {
             AggregateFunc::Count => false,
+            // Use the nullability of the underlying column being aggregated, not the Records wrapping it
+            AggregateFunc::StringAgg { .. } => match input_type.scalar_type {
+                // The outer Record wraps the input in the first position, and any ORDER BY expressions afterwards
+                ScalarType::Record { fields, .. } => match &fields[0].1.scalar_type {
+                    // The inner Record is a (value, separator) tuple
+                    ScalarType::Record { fields, .. } => fields[0].1.nullable,
+                    _ => unreachable!(),
+                },
```

Comment:
Ugh, this is so mind-bending. Do we not need to do this for the other order-sensitive aggregates?

Reply:
I've checked all the other cases, and they never return nulls, so they are correct!
- `jsonb_agg`: doesn't ignore NULLs, so it returns a JSON array of NULLs.
- `array_agg`: doesn't ignore NULLs, so it returns an array of NULLs.
- `list_agg`: doesn't ignore NULLs, so it returns a list of NULLs.
- `jsonb_agg`: doesn't ignore NULLs, but it should actually _fail_ when the key is null, going to open a separate issue for it later
- `row_number`: not relevant, doesn't process the value

Technically, they should maybe be special cased like `count`, but I wanted to keep this PR more focused.